addons/website/static/src/snippets/s_rating/options.js

```diff
@@ -0,0 +1,184 @@
+                $('<i/>')
+            );
+            this._saving = false;
+            dialog.on('save', this, function (attachments) {
+                this._saving = true;
+                const $icons = this.$target.find('.rating_icon');
+                _.each($icons, el => {
+                    if (el.dataset.iconActive === params.customActiveIcon) {
+                        $(el).removeClass().addClass('rating_icon fa-fw ' + attachments.className);
+                    }
+                });
+                this.faClassActiveCustomIcons = $icons.filter('[data-icon-active="true"]').attr('class');
+                this.faClassInactiveCustomIcons = $icons.filter('[data-icon-active="false"]').attr('class');
+                this.$target.attr('data-active-custom-icon', this.faClassActiveCustomIcons);
+                this.$target.attr('data-inactive-custom-icon', this.faClassInactiveCustomIcons);
+                this.$target.attr('data-icon', 'custom');
+                this.iconType = 'custom';
+                resolve();
+            });
+            dialog.on('closed', this, function () {
+                if (!this._saving) {
+                    resolve();
+                }
+            });
+            dialog.open();
+        });
+    },
+
+    /**
+     * Set the number of active icons.
+     *
+     * @see this.selectClass for parameters
+     */
+    activeIconsNumber: function (previewMode, widgetValue, params) {
+        if (widgetValue > 0) {
+            this.$target.find('.rating_icon').attr('data-icon-active', 'false');
+            this.$target.find('.rating_icon:nth-child(-n+' + widgetValue + ')').attr('data-icon-active', 'true');
+            this._renderIcons();
+        } else {
+            return;
+        }
+    },
+
+    /**
+     * Set the total number of icons.
+     *
+     * @see this.selectClass for parameters
+     */
+    totalIconsNumber: function (previewMode, widgetValue, params) {
+        let $icons = this.$target.find('.rating_icon');
+        const totalIcons = $($icons).length;
+        if (widgetValue < 1) {
+            return;
+        } else if (widgetValue > totalIcons) {
+            let i = 0;
+            const count = widgetValue - totalIcons;
+            for (i = 0; i < count; i++) {
+                $($icons[totalIcons - 1]).clone().insertAfter($($icons[totalIcons - 1])).before(' ');
+            }
+        } else if (widgetValue < totalIcons) {
+            this.$target.find(".rating_icon:nth-last-child(-n+" + (totalIcons - widgetValue) + ")").remove();
+        }
+    },
+
+    /**
+     * Set icon color.
+     *
+     * @see this.selectClass for parameters
+     */
+    setIconColor: function (previewMode, widgetValue, params) {
+        this.activeIconColor = this._ensureCssColor(this.$target.attr('data-active-icon-color'));
+        this.inactiveIconColor = this._ensureCssColor(this.$target.attr('data-inactive-icon-color'));
+        this._renderIcons();
+    },
```

Comment:
This method does nothing ?
EDIT: ok I understood -> you cannot rely on the order in which the methods will be called, so here it is by chance that the `selectDataAttribute` is executed before.

Also, you cannot use that `_ensureCssColor` in a snippet option (Romain used it in its widget on each page load, not in an option) :/ If the user selects the "delta" color, you transform that into an hexadecimal and then the user is stuck with hardcoded colors in its DOM and those won't change if the user changes "delta" someday. Here you have to use the text-color classes in these cases. I understand the problem that if the user selects 5/5, you won't be able to remember the inactive color. Maybe you can split the layout in two divs: a left part which contains the active icons and a right part which contains the inactive ones. You can then apply the color on the div instead of the icons. Then, you won't have the data-active-icon attributes on icons, just the icons they use and in which div they are placed. What do you think ? I let you check and adapt if possible before continuing the review :+1: